### ISSUE TITLE
Dynamically constrain range sliders

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega-lite#4631]
Suppose you wished to make an interval selection with a pair of range sliders, one specifying a minimum value, the other a maximum value. We can currently do this with something like this:

```javascript
"selection": {
    "minThreshold": {
      "type": "single",
      "fields": ["crimes"],
      "bind": {
        "crimes": { "input": "range", "name": "Min", "max": 1000}
      }
    },
    "maxThreshold": {
      "type": "single",
      "fields": ["crimes"],
      "bind": {
        "crimes": {"input": "range", "name": "Max", "max": 1000}
      }
    }
  }
```

However, this can result in the `minThreshold` range input having a greater value than the `maxThreshold` range input. It would be great if it were possible to specify the min, max and step of a range input as a dynamic property of other selections (as you can in Vega). This is common in temporal range selections where increasing the _start_ dateTime automatically shifts the _end_ dateTime if _start_ would otherwise be >= _end_.

Perhaps via a specification such as :

```javascript
"selection": {
    "minThreshold": {
      "type": "single",
      "fields": ["crimes"],
      "bind": {
        "crimes": { "input": "range", "name": "Min", "max": "maxThreshold"}
      }
    },
    "maxThreshold": {
      "type": "single",
      "fields": ["crimes"],
      "bind": {
        "crimes": {"input": "range", "name": "Max", "min" : "minThreshold", "max": 1000}
      }
    }
  }
```

**Comment by jwoLondon** [comment]
On reflection, it would probably make more sense to be able to bind the value of the range input rather than its min/max, although the binding should only be applied conditionally, perhaps via a predicate expression (e.g. `minThreshold >= maxThreshold`)?

**Comment by domoritz** [comment]
Just as an alternative for now, you could use an interval selection on a timeline to get a range slider.

**Comment by jwoLondon** [comment]
Yes, I realise interval selections can select within bounds, but that only works if the field over which to make the selection is encoded with a position channel. The advantage of range inputs is they can be bound to non-positional fields.

**Comment by domoritz** [comment]
I'll mark this as a feature request.
